### DISCUSSION TITLE
bundle for k3s airgap images

### DISCUSSION THREAD

**Discussion by poblin-orange** [kairos-io/kairos#1283] [Ideas]
Hello,
to help with disconnected  environment, k3s provides a package of core images
- https://github.com/k3s-io/k3s/releases/download/v1.26.3%2Bk3s1/k3s-airgap-images-amd64.tar

This must be set in /var/lib/rancher/k3s/agent/images/, that way k3s core feature (klipper traeffik, coredns) can start without docker registry connectivity

These looks like a simple bundle, could be part of core Kairos bundles ?

**Comment by Itxaka** [comment]
Hi @poblin-orange !

There is already docs on how to do this: https://kairos.io/docs/examples/airgap/ and get an iso with the bundled images inside it, ready for airgapped environments.

Would that work for you?

**Reply by poblin-orange** [reply]
Hi @Itxaka 
Thks for the link, i will try it.
My point was also to include k3s airgap packages as part of core kairos (kairos offers packaged OS+k3s runtime image, these airgap k3s image are versionned according to k3s runtime)

**Reply by mudler** [reply]
Hi @poblin-orange :wave: 

Thanks! I think this is good food for thought, could you please file an issue so we can track it in our sprints? I think that could increase image size, however, I see we could add probably some automation and create airgapped bundles so to be consumed easily without having to build them manually in our community-bundle repository

**Reply by poblin-orange** [reply]
Thks @mudler . I tried to express the point in https://github.com/kairos-io/kairos/issues/1289
